docs/bots-guide.md

```diff
@@ -443,6 +443,103 @@ bot_handler.update_message(dict(
   development environment hosted on your computer, use
   `localhost:9991`
 
+### Writing tests for bots
+
+Bots in the [`python-zulip-api`](https://github.com/zulip/python-zulip-api/tree/master/zulip_bots/zulip_bots/bots)
+repository should have unittests. This section will explain the basics of writing a test suite for a bot.
+
+*Unittests for bots make heavy use of mocking. If you want to get comfortable with mocking, mocking strategies, etc.
+ you should check out our [mocking guide](testing-with-django.html#testing-with-mocks).*
+
+#### A simple example
+
+ Let's have a look at a simple test suite for the [`helloworld`](
+ https://github.com/zulip/python-zulip-api/tree/master/zulip_bots/zulip_bots/bots/helloworld) bot
+ (the actual test is written slightly more compact).
+
+     #!/usr/bin/env python
+
+    from __future__ import absolute_import
+    from __future__ import print_function
+
+    from six.moves import zip
```

Comment:
Does this actually use `zip`?  Seems like we could remove it and simplify this example.

Reply:
Whoops, yeah,  the original test uses zip, making `check_expected_responses` a one-liner.